domains/runtime/evm/src/lib.rs

```diff
@@ -897,9 +899,21 @@ fn is_xdm_mmr_proof_valid(ext: &<Block as BlockT>::Extrinsic) -> Option<bool> {
 /// Returns a valid Sudo call.
 /// Should extend this function to limit specific calls Sudo can make when needed.
 fn is_valid_sudo_call(encoded_ext: Vec<u8>) -> bool {
-    UncheckedExtrinsic::decode(&mut encoded_ext.as_slice()).is_ok()
+    let Ok(extrinsic) = UncheckedExtrinsic::decode(&mut encoded_ext.as_slice()) else {
+        return false;
+    };
+
+    match extrinsic.0.function {
+        // Only private EVMs accept contract creation allow list changes.
+        RuntimeCall::EVMNoncetracker(
+            pallet_evm_tracker::Call::set_contract_creation_allowed_by { .. },
+        ) => pallet_evm_tracker::Pallet::<Runtime>::is_private_evm_domain(),
+        // TODO: extend this function to limit specific calls Sudo can make when needed.
+        _ => true,
+    }
```

Comment:
IMHO we should keep `is_valid_sudo_call` simple to only check if the `encoded_ext` is indeed an extrinsic of the evm runtime, any further check should be performed in the execution of the extrinsic itself.

In this specific case, using domain sudo to call `set_contract_creation_allowed_by` will fail due to the `ensure_none(origin)` check during execution in the domain chain.

Reply:
Aha, I see this is `set_contract_creation_allowed_by` not `inherent_set_contract_creation_allowed_by`, let's move the discussion there.